File: output/CodeGen/avatar/Java/Python/atcoder_ABC150_E.py
def main ():
    n = int(input())
    sa = input().split()
    c = [int(sa[i]) for i in range(n)]
    mod = 1000000007
    if n == 1:
        print(c[0] * 2 % mod)
        return
    sa = input().split()
    c = [int(sa[i]) for i in range(n)]
    b = 2 ** n
    a = 2 ** (n - 2)
    ans = 0
    for i in range(2, n + 1):
        val = a * i % mod
        val *= c[n + 1 - i]
        val %= mod
        ans += val
        ans %= mod
    ans *= b
    ans %= mod
    print(ans)

File: output/ChatGPT/avatar/Python/Java/atcoder_ABC150_E.java
public class atcoder_ABC150_E {
    public static void main(String[] args) {
        int mod = (int) 1e9 + 7;
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int[] c = new int[n];
        for (int i = 0; i < n; i++) {
            c[i] = scanner.nextInt();
        }
        Arrays.sort(c);
        long b = pow(2, 2 * n - 2, mod);
        long a = 2 * b % mod;
        long ans = 0;
        for (int i = 0; i < n; i++) {
            ans += c[i] * (a + i * b);
            ans %= mod;
        }
        System.out.println(ans);
    }
    
    public static long pow(long base, long exponent, int mod) {
        long result = 1;
        while (exponent > 0) {
            if (exponent % 2 == 1) {
                result = result * base % mod;
            }
            base = base * base % mod;
            exponent /= 2;
        }
        return result;
    }
}